slicedimage/backends/_caching.py

```diff
@@ -1,31 +1,37 @@
 from __future__ import absolute_import, division, print_function, unicode_literals
 
-import os
+import io
+
+from diskcache import Cache
 
 from ._base import Backend
 
+size_limit = 5e9
+
 
 class CachingBackend(Backend):
+
     def __init__(self, cacheroot, authoritative_backend):
         self._cacheroot = cacheroot
         self._authoritative_backend = authoritative_backend
+        self.cache = Cache(cacheroot, size_limit=int(size_limit))
 
-    def read_file_handle_callable(self, name, checksum_sha1=None, seekable=False):
-        # TODO: (ttung) This is a very very primitive cache.  Should make this more robust with
-        # evictions and all that good stuff.
-        cachedir = os.path.join(self._cacheroot, checksum_sha1[0:2], checksum_sha1[2:4])
-        cachepath = os.path.join(cachedir, checksum_sha1)
-        if not os.path.exists(cachepath):
-            os.makedirs(cachedir)
-            with open(cachepath, "w") as dfh, \
-                    self._authoritative_backend.read_file_handle(name, checksum_sha1) as sfh:
-                while True:
-                    data = sfh.read(128 * 1024)
-                    if len(data) == 0:
-                        break
-                    dfh.write(data)
-
-        return lambda: open(cachepath, "r")
+    def read_file_handle_callable(self, name, checksum_sha256=None, seekable=False):
```

Comment:
I know that most of the code here doesn't have docstrings, but now is as good a time as any to start :)